### ISSUE TITLE
[BUG]: Typescript Compilation Error

### ISSUE THREAD

**Issue by aronmalkine** [paypal/react-paypal-js#281]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### 🐞 Describe the Bug

Webpack compilation fails with multiple typescript errors on /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts. This prevents automatic reloading of react browser window and prevents us from deploying.

### 😕 Current Behavior

When a project file change is detected, Webpack compilation fails with the above errors. 

Opening the file at issue, /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts, visual studio code highlights the same issues in the codebase.

### 🤔 Expected Behavior

When a project file change is detected, Webpack compilation should succeed.

There should be no errors highlighted in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts when open in visual studio code.

### 🔬 Minimal Reproduction

```markdown
1. Install react-paypal-js
2. Import with:
 import { PayPalScriptProvider, PayPalButtons } from '@paypal/react-paypal-js';
3. Add PayPal button in tsx file with
        <PayPalScriptProvider options={paypalOptions}>
            <PayPalButtons
                style={{ layout: 'horizontal' }}
                createOrder={createOrder}
                onApprove={handleApproval}
                fundingSource={'paypal'}
            />
            <Cancel onClick={handleCancel}>Cancel</Cancel>
        </PayPalScriptProvider>
4. Save file
5. Automatic recompile fails with above errors
```

### 🌍 Environment

```markdown
| Software         | Version(s)   |
| ---------------- | ----------   |
| react-paypal-js  | 7.8.1        |
| Browser          | N/A          |
| Operating System | OSx Monterey |
```

### Relevant log output

```shell
ℹ ｢wdm｣: Compiling...
✖ ｢wdm｣: Hash: f7b4617484465d494225
Version: webpack 4.44.2
Time: 8056ms
Built at: 06/07/2022 8:21:51 PM
                                  Asset      Size  Chunks                               Chunk Names
              0.f7b4617484465d494225.js    14 MiB       0  [emitted] [immutable]
   9abef9a193d343e788de.hot-update.json  46 bytes          [emitted] [immutable] [hmr]
                             index.html  2.09 KiB          [emitted]
main.9abef9a193d343e788de.hot-update.js   105 KiB    main  [emitted] [immutable] [hmr]  main
           main.f7b4617484465d494225.js  59.4 MiB    main  [emitted] [immutable]        main
 + 131 hidden assets
Entrypoint main = main.f7b4617484465d494225.js main.9abef9a193d343e788de.hot-update.js
[./src/react/fragments/ActionTray.tsx] 10.6 KiB {main} [built]
[./src/react/menuUI/modals/TipsModal.tsx] 21.9 KiB {main} [built]
    + 1937 hidden modules
    
ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts
[tsl] ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts(10,11)
      TS1110: Type expected.

ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts
[tsl] ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts(12,1)
      TS1128: Declaration or statement expected.

ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts
[tsl] ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts(12,3)
      TS1109: Expression expected.

ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts
[tsl] ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts(13,45)
      TS1005: ',' expected.

ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts
[tsl] ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts(14,58)
      TS1005: ',' expected.

ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts
[tsl] ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts(16,54)
      TS1005: ',' expected.

ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts
[tsl] ERROR in /usr/src/app/node_modules/@paypal/react-paypal-js/dist/types/types/scriptProviderTypes.d.ts(17,35)
      TS1005: ',' expected.
ℹ ｢wdm｣: Failed to compile.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aronmalkine** [comment]
Resolved by upgrading typescript from version 4.0.5 to version v4.7.3.

**Comment by borodovisin** [comment]
Hi @aronmalkine, can you share the typescript version are you using?
